src/test/groovy/org/gradle/android/TaskAvoidanceTest.groovy

```diff
@@ -0,0 +1,55 @@
+package org.gradle.android
+
+import org.gradle.util.GradleVersion
+import org.gradle.util.VersionNumber
+import spock.lang.Unroll
+
+class TaskAvoidanceTest extends AbstractTest {
+    @Unroll
+    def "tasks are avoided with #gradleVersion and Android plugin #androidVersion"() {
+        assert gradleVersion instanceof GradleVersion
+        assert androidVersion instanceof VersionNumber
+
+        println "> Using Android plugin $androidVersion"
+        println "> Running with $gradleVersion"
+
+        SimpleAndroidApp.builder(temporaryFolder.root, cacheDir)
+                .withAndroidVersion(androidVersion)
+        .withKotlinDisabled()
+                .build()
+                .writeProject()
+
+        def originalSettings = file('build.gradle').text
+        file('build.gradle').text = """
+            ${originalSettings}
+
+            allprojects {
+              tasks.configureEach {
```

Comment:
I would only do these for type `AbstractCompile` and then in the assertion, ensure that there is no output that contains "configuring...".

Reply:
Good point, but maybe it's good to keep this as is in order to test that the Android Gradle Plugin does not regresses in any way in terms of task configuration avoidance. One would argue that we shouldn't be testing this for Google, but that's basically what we do with the `CrossVersionOutcomeAndRelocationTest` by testing all the tasks in the Android Gradle Plugin.

Reply:
Ah, I see what you mean though after reading the comment below. The test is more thorogh. Will change.